app/models/problem_check.rb

```diff
@@ -106,8 +133,31 @@ def call
     raise NotImplementedError
   end
 
+  def run
+    problems = call
+
+    yield(problems) if block_given?
+
+    next_run_at = perform_every&.from_now
+
+    if problems.empty?
+      tracker.no_problem!(next_run_at:)
+    else
+      tracker.problem!(
```

Comment:
What happens if there are multiple problems with different messages?

Reply:
I have been skirting around that question. The only case I found was what seemed like two problem checks co-located in one place, so I'm not sure we should add support for multiple problems proactively. Unless you have a compelling case.